### PR TITLE
Avoid unnecessary copies in VotingConfiguration serialization

### PR DESCRIPTION
**Pull request by DaveCTurner**
These objects take a route to/from their wire representation via an array, which is unnecessary. It's the same bytes on the wire as a list, so we can just use a list instead.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationMetadata.java [hunk 1]
```diff
@@ -337,12 +336,12 @@ public VotingConfiguration(Set<String> nodeIds) {
         }
 
         public VotingConfiguration(StreamInput in) throws IOException {
-            nodeIds = Collections.unmodifiableSet(Sets.newHashSet(in.readStringArray()));
+            nodeIds = Set.copyOf(in.readStringList());
```
Comment (idegtiarenko) [inline server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationMetadata.java:339]: NIT:
```suggestion
            nodeIds = Set.of(in.readStringArray());
```
I believe this way we would not need to create intermediate list. Array would still be copied.

**Review by idegtiarenko — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationMetadata.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/cluster/coordination/CoordinationMetadata.java]: Ah right yes even nicer.

**Review by fcofdez — APPROVED** [review]
LGTM

**Comment by DaveCTurner** [comment]
I'm ignoring the failure in the `full-bwc` tests - we don't run this suite normally anyway, it passed for all versions except 8.7.0, and the failure there looks unrelated.

**Pull request merged**